### PR TITLE
Add goreleaser

### PR DESCRIPTION
**Pull request by arbourd**
Adds release workflow that runs GoReleaser, creating archives for atmo
and atmo-proxy on tagged releases.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        - goos: linux
+          ldflags:
+            - -extldflags "-static"
+  - <<: *build_defaults
+    id: atmo-proxy
+    binary: atmo-proxy
+    tags: proxy
+
+changelog:
+  skip: true
```
Comment (rnpridgeon) [inline .goreleaser.yml:26]: any reason not to introduce a changelog generator?

**Review by rnpridgeon — APPROVED** [review]
seems fine to me though I am curious if there is a reason we are intentionally avoiding a change log generator

File: .goreleaser.yml (hunk 1 above)
Comment (cohix) [reply .goreleaser.yml:22]: We don't use conventional commits everywhere yet

Comment (arbourd) [reply .goreleaser.yml:22]: We generate our own AFAIK

**Pull request merged**